packager/status/status.h

```diff
@@ -153,7 +153,7 @@ class SHAKA_EXPORT Status {
   // generated copy constructor and assignment operator.
 };
 
-SHAKA_EXPORT std::ostream& operator<<(std::ostream& os, const Status& x);
+std::ostream& operator<<(std::ostream& os, const Status& x);
```

Comment:
I had to remove SHAKA_EXPORT from this line to fix the duplicate symbol errors when building shared library mode on Windows. Not sure whether this is going to break any users of shared libpackager?

Reply:
Hrm... interesting.  I can't say what people are or aren't using with the library.  But it seems like a useful ability to output a Status object for logging or debugging.

Sometimes duplicate symbol errors occur when a definition is inline in a header, but I see that isn't the case here.  Could status.o be included in multiple shared libraries, or both in the shared library and an executable?

Reply:
The implementation is fairly trivial though, it just calls .ToString() on the status object. 

I did try to figure out why it's breaking though, I checked to see whether packager executable and libpackager both pull in status for example (or the test suite) and didn't have any luck. It's also only breaking on Windows (shared) specifically, none of the other platforms have this issue. It seems related to the dllimport / dllexport tagging going on here but I'm not too familiar with these Microsoft extensions.

Perhaps someone that knows more about Windows development can shed some light on it. But since the implementation of << is fairly trivial I think it's ok to drop this for now until we can find a proper solution.

Reply:
Agreed.  I wanted to do a major-version bump with the first cmake release anyway, since I don't know that we can guarantee ABI compat after all these changes.